Src/main.c

```diff
@@ -112,9 +117,17 @@ int main(void)
 
   /* USER CODE BEGIN 3 */
 
+  HAL_ADC_PollForConversion(&hadc, 20); //Laye - polls potentiometer for analog values 
+  converted_val = HAL_ADC_GetValue(&hadc); //Laye - return value of ADC 
+  compare_val = (converted_val/4096)*12000; // Laye - scaling 12 bit return value according to period of 12000
+  __HAL_TIM_SET_COMPARE(&htim16, TIM_CHANNEL_1, compare_val);  //Laye - setting compare register to change duty cycle
+  HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1); //Laye - starting PWM
```

Comment:
This only needs to be called once and the PWM will keep running (similar to the HAL_ADC_Start function), so you can move it out of the loop.